sources/rest_api/__init__.py

```diff
@@ -259,12 +250,12 @@ def paginate_resource(
                 hooks: Optional[Dict[str, Any]],
                 client: RESTClient = client,
                 incremental_object: Optional[Incremental[Any]] = incremental_object,
-                incremental_param: IncrementalParam = incremental_param,
+                incremental_param: Optional[IncrementalParam] = incremental_param,
             ) -> Generator[Any, None, None]:
                 if incremental_object:
-                    params[incremental_param.start] = incremental_object.last_value
-                    if incremental_param.end:
-                        params[incremental_param.end] = incremental_object.end_value
+                    params = _set_incremental_params(
+                        params, incremental_object, incremental_param, incremental_cursor_transform
```

Comment:
`incremental_cursor_transform` is always `None` at runtime outside of the unit tests. 
Is extending the function signatures the only way?

Reply:
Most likely, there's an alternative way, but I haven't discovered it yet. Let's use explicit params and experiment with the alternative implementation as a follow-up.

Reply:
excellent, thank you for the clarification.